test/index.spec.ts

```diff
@@ -60,22 +60,44 @@ describe('resin-semver', () => {
 			expect(semver.compare('2.0.0+rev6', 'Resin OS 2.0.0+rev3')).to.equal(1);
 			expect(semver.compare('2.0.0+rev2', 'Resin OS 2.0.0+rev5')).to.equal(-1);
 			expect(semver.compare('2.0.0+rev3', '2.0.0+rev3')).to.equal(0);
+			expect(semver.compare('2.0.0+rev1', '2.0.0+rev11')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev2', '2.0.0+rev11')).to.equal(-1);
 		});
 
 		it('should correctly compare ".dev" versions', () => {
-			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(1);
+			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(-1);
 			expect(semver.compare('2.0.0', '2.0.0-dev')).to.equal(1);
-			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(-1);
+			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(1);
 			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6.dev')).to.equal(0);
-			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(1);
```

Comment:
As above: dev should be less than non-dev, with these changes most of these are now flipped, incorrectly.

Reply:
Your comment makes sense, and goes against the previous "more specific is higher ranked" argument. I guess it would be reconciled if it would mean that versions that don't have `.dev` or any other non-`.prod` extension are treated as `.prod`, and thus ranking higher?
(let me know if it makes sense)